uaa/src/test/java/org/cloudfoundry/identity/uaa/orchestrator/UaaOrchestratorClaimTest.java

```diff
@@ -0,0 +1,301 @@
+
+    public static final String ZONES_URI = "/orchestrator/zones";
+
+    private UaaRequest uaaRequest;
+
+    protected static HttpHeaders headers;
+    Random random = new Random();
+
+    private  final String UAA_SUBDOMAIN = "rest-uaa-subdomain"+random.nextInt(1000);
+
+    private static final String UAA_ADMIN_CLIENT_SECRET = "ZHVtbXlBZG1pblNlY3JldDEyMyMh";
+
+    private static final String UAA_CLAIM_NAME_PREFIX = "uaa-it-";
+    @Value("${uaa.url}")
+    private String uaaUrl;
+
+    @Value("${uaa.orchestrator.clientId}")
+    private String clientId;
+
+    @Value("${uaa.orchestrator.clientSecret}")
+    private String clientSecret;
+
+    @Value("${kubernetes.config.env}")
+    private String environment;
+
+    @Autowired
+    private OrchestratorUtil orchestratorUtil;
+
+    protected static OAuth2RestTemplate trustedIssuerValidScopesRestTemplate;
+
+    @Value("${kubernetes.config.secondary-namespace:iam-dev}")
+    private String namespace;
+
+
+    private Integer uaaInstanceCount = random.nextInt(1000);
+    ;
+
+
+    // Below regex UMS_EKS_ENV_URI_REG will match all UMS EKS Env uris. Examples, https://user-management-int.pss.dev.usw02.15.energy
+    // https://user-management-dev.pss.dev.usw02.15.energy, https://user-management-stg.pss.dev.usw02.15.energy
+    private static String UMS_EKS_ENV_URI_REG =
+            "(https://[-a-zA-Z0-9]*(\\.uaa-qa\\.pss\\.dev\\.usw02\\.15\\.energy){1,1})";
+
+    @Before
+    public void setupDataBeforeEachTestRun() {
+        uaaRequest = getUaaRequest();
+    }
+
+    // region Positive Tests
+    @Test
+    public void testCreateZone() throws Exception {
+
+        OrchestratorZoneResponse getZone = null;
+        String zoneName = null;
+        String claimName =
+                (UAA_CLAIM_NAME_PREFIX +/* testConfig.getTestRunIdentifier() + */ "-" + (++uaaInstanceCount)).toLowerCase();
+
+        Claim claim = null;
+        boolean isClaimCreated = false;
+        boolean isZoneCreated = false;
+        try {
+            isClaimCreated = createZone(claimName, uaaRequest);
+            Thread.sleep(120000);
+            claim = getZone(claimName, ServiceName.UAA);
+            assertNotNull(claim);
+            assertEquals(claimName, claim.getMetadata().getName());
+
+            zoneName = namespace + "-" + claimName;
+            String getResponse = getZoneByName(zoneName, HttpStatus.OK);
+            getZone = OBJECT_MAPPER.readValue(getResponse, OrchestratorZoneResponse.class);
+
+            assertNotNull(getResponse);
+            assertNotNull(getZone);
+            assertEquals(zoneName, getZone.getName());
+            assertNotNull(getZone.getConnectionDetails().getDashboardUri());
+            assertNotNull(getZone.getConnectionDetails().getUri());
+            assertNotNull(getZone.getConnectionDetails().getIssuerId());
+            assertEquals(uaaRequest.getSubdomain(), getZone.getConnectionDetails().getSubdomain());
```

Comment:
Also add validations for asserting zone in connection details. Refer sample get response here https://devcloud.swcoe.ge.com/devspace/display/RMOCM/Orchestrator+UAA+Zone+provisioning+API#OrchestratorUAAZoneprovisioningAPI-GETAPI

Reply:
Done